workspaces/src/network/result.rs

```diff
@@ -53,3 +55,33 @@ impl From<FinalExecutionOutcomeView> for CallExecutionDetails {
         }
     }
 }
+
+/// The result from a call into a View function. This contains the contents or
+/// the results from the view function call itself. The consumer of this object
+/// can choose how to deserialize its contents.
+#[non_exhaustive]
+pub struct ViewResultDetails {
+    /// Our result from our call into a view function.
+    pub result: Vec<u8>,
+    /// Logs generated from the view function.
+    pub logs: Vec<String>,
+}
+
+impl ViewResultDetails {
+    pub fn json<T: serde::de::DeserializeOwned>(&self) -> anyhow::Result<T> {
+        serde_json::from_slice(&self.result).map_err(Into::into)
+    }
+
+    pub fn borsh<T: borsh::BorshDeserialize>(&self) -> anyhow::Result<T> {
```

Comment:
I think docs on these would be very useful. It's very unclear by the names `json` and `borsh` that this is trying to deserialize the result imo.